src/config/createForm.controller.js

```diff
@@ -30,7 +28,27 @@ export default ({ form, formConfirmation }) => {
     form.submit();
   });
 
+  // in a form submission, it will either fire `submitDone` or `submitError`, after getting the response from the formio api.
   form.on("submitDone", () => {
-    if (formConfirmation) window.location.href = formConfirmation;
+    pushDataLayer({
+      event: "formio-submission",
+      submissionsUrl: `form.io: ${form.formio.submissionsUrl}`,
+      "formio-name": form._form.title,
+      "formio-version": form._form.modified,
+    });
+
+    if (formConfirmation)
+      setTimeout(() => {
+        window.location.href = formConfirmation;
+      }, 500);
+  });
+
+  form.on("submitError", (error) => {
+    pushDataLayer({
+      event: "ngErrorEvent",
+      ngErrorLocation: form._form.title,
+      ngErrorMsg: "submission error",
```

Comment:
shouldn't msg with the error?.message || "submission error" ?

Reply:
yep that's right, just confirmed with steve, will put the message in ngErrorMsg and leave ngErrorStack empty